Mathlib/Data/Complex/Order.lean

```diff
@@ -125,3 +125,31 @@ lemma monotone_ofReal : Monotone ofReal' := by
   simp only [ofReal_eq_coe, real_le_real, hxy]
 
 end Complex
+
+namespace Mathlib.Meta.Positivity
+open Lean Meta Qq Complex
+open scoped ComplexOrder
+
+private alias ⟨_, ofReal_pos⟩ := zero_lt_real
+private alias ⟨_, ofReal_nonneg⟩ := zero_le_real
+private alias ⟨_, ofReal_ne_zero_of_ne_zero⟩ := ofReal_ne_zero
+
+/-- Extension for the `positivity` tactic: `Complex.ofReal` is positive/nonnegative/nonzero if its
+input is. -/
+@[positivity Complex.ofReal' _]
+def evalComplexOfReal : PositivityExt where eval {u α} _ _ e := do
+  match u, α, e with
+  | 0, ~q(ℂ), ~q(Complex.ofReal' $a) =>
+    assumeInstancesCommute
+    match ← core q(inferInstance) q(inferInstance) a with
+    | .positive pa => return .positive q(ofReal_pos $pa)
+    | .nonnegative pa => return .nonnegative q(ofReal_nonneg $pa)
+    | .nonzero pa => return .nonzero q(ofReal_ne_zero_of_ne_zero $pa)
+    | _ => return .none
+  | _, _ => throwError "not Complex.ofReal'"
```

Comment:
We can implement it for multiple morphisms without quite duplicating everything, like this:
```suggestion
@[positivity (Complex.ofReal' _), (Complex.ofReal _)]
def evalComplexOfReal : PositivityExt where eval {u α} _ _ e := do
  match u, α, e with
  | 0, ~q(ℂ), ~q(Complex.ofReal' $a) =>
    assumeInstancesCommute
    match ← core q(inferInstance) q(inferInstance) a with
    | .positive pa => return .positive q(ofReal_pos $pa)
    | .nonnegative pa => return .nonnegative q(ofReal_nonneg $pa)
    | .nonzero pa => return .nonzero q(ofReal_ne_zero_of_ne_zero $pa)
    | _ => return .none
  | 0, ~q(ℂ), ~q(Complex.ofReal $a) =>
    assumeInstancesCommute
    match ← core q(inferInstance) q(inferInstance) a with
    | .positive pa => return .positive q(ofReal_pos $pa)
    | .nonnegative pa => return .nonnegative q(ofReal_nonneg $pa)
    | .nonzero pa => return .nonzero q(ofReal_ne_zero_of_ne_zero $pa)
    | _ => return .none
  | _, _ => throwError "not Complex.ofReal'"
```

However, I can't figure out how to pull out the duplication into a local definition. It would be something like:
```suggestion
@[positivity (Complex.ofReal' _), (Complex.ofReal _)]
def evalComplexOfReal : PositivityExt where eval {u α} _ _ e := do
  match u, α, e with
  | 0, ~q(ℂ), ~q(Complex.ofReal' $a) => go a ⟨⟩
  | 0, ~q(ℂ), ~q(Complex.ofReal $a) => go a ⟨⟩
  | _, _ => throwError "not Complex.ofReal'"
  where
    go {e} (a : Q(ℝ)) (h : $e =Q ofReal $a) : MetaM (Strictness _ _ e) := do
    match ← core q(inferInstance) q(inferInstance) a with
    | .positive pa => return .positive q(ofReal_pos $pa)
    | .nonnegative pa => return .nonnegative q(ofReal_nonneg $pa)
    | .nonzero pa => return .nonzero q(ofReal_ne_zero_of_ne_zero $pa)
    | _ => return .none
```

Reply:
I've committed your code with a todo to deduplicate